### PR TITLE
cli: remove dependence on less

### PR DESCRIPTION
**Pull request by oliver-sanders**
See https://github.com/cylc/cylc-flow/issues/2957#issuecomment-535222497

These changes partially address #2957 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (meh).
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
Today-I-Learned about the `$PAGER` [environment variable](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/man.html).

>Determine an output filtering command for writing the output to a terminal. Any string acceptable as a command_string operand to the sh -c command shall be valid. When standard output is a terminal device, the reference page output shall be piped through the command. If the PAGER variable is null or not set, the command shall be either more or another paginator utility documented in the system documentation.

Thanks!

**Comment by kinow** [comment]
One Travis error, not sure if related.

```
./tests/restart/00-pre-initial.t                                (Wstat: 0 Tests: 4 Failed: 2)
  Failed tests:  3-4
```

**Comment by oliver-sanders** [comment]
> One Travis error, not sure if related.

hmm, runs OK in my environment, kicking Travis.

File: bin/cylc [hunk 1]
```diff
@@ -165,6 +165,16 @@ run_cylc_command() {
     exec "${CYLC_HOME_BIN}/cylc-${UTIL}" "$@"
 }
 
+pager () {
+    if [[ -n $PAGER ]]; then
```
Comment (hjoliver) [inline bin/cylc]:
```suggestion
    if [[ -n ${PAGER:-} ]]; then
```
Avoid `cylc-flow/venv/bin/cylc: line 169: PAGER: unbound variable`

File: bin/cylc [hunk 2]
```diff
@@ -165,6 +165,16 @@ run_cylc_command() {
...
+        echo "$PAGER"
+    elif command -v less; then
```
Comment (hjoliver) [inline bin/cylc]:
```suggestion
    elif command -v less > /dev/null; then
```

This line prints to stdout, resulting (with the next line) in (e.g.) `/usr/bin/less less`

File: bin/cylc-jobscript [hunk 3]
```diff
@@ -51,6 +51,16 @@ elif sys.argv[1] in ['-e', '--edit']:
...
+pager () {
+    if [[ -n $PAGER ]]; then
+        echo "$PAGER"
+    elif command -v less; then
+        echo less
+    else
+        echo cat
+    fi
+}
+
```
Comment (hjoliver) [inline bin/cylc-jobscript]: Ditto comments above.

(Should really be in a common shell library) (but ... see final review comment...)

**Comment by oliver-sanders** [comment]
> I think this will all be rendered pointless by @sgaist's PR

Hopefully!

**Comment by kinow** [comment]
(just waiting for travis OK to merge it :+1: )

**Comment by kinow** [comment]
Tried the same test that just failed on [this travis build](https://travis-ci.org/cylc/cylc-flow/jobs/601165824) and it failed for me too.

```
$ ./etc/bin/run-functional-tests.sh -v ./tests/cylc-insert/13-family-submit-num.t

ok 1 - 13-family-submit-num-validate
not ok 2 - 13-family-submit-num-run
not ok 3 - 13-family-submit-num-bar-02

    stdout and stderr stored in: /tmp/kinow/cylctb-20191022T093711Z/cylc-insert/13-family-submit-num
    suite logs can be found under: /home/kinow/cylc-run/cylctb-20191022T093711Z/cylc-insert/13-family-submit-num/log/suite/
Failed 2/3 subtests 
[22:39:15]

Test Summary Report
-------------------
./tests/cylc-insert/13-family-submit-num.t (Wstat: 0 Tests: 3 Failed: 2)
  Failed tests:  2-3
Files=1, Tests=3, 123 wallclock secs ( 0.03 usr  0.00 sys +  1.39 cusr  0.22 csys =  1.64 CPU)
Result: FAIL

```

There is another build that failed too, and in that one, the `./tests/restart/00-pre-initial.t` test failed again.

**Comment by kinow** [comment]
(not kicking Travis so you can have a look at the errors @oliver-sanders )

**Comment by oliver-sanders** [comment]
What the heck! Tested "in rage" locally, could not reproduce any of those failures. Rebased and pushed...

**Comment by oliver-sanders** [comment]
All passed first time, hmmmm.

**Pull request merged**